### PR TITLE
Convert VersionCompat into a library, and introduce a CompareVersions tool

### PR DESCRIPTION
**Pull request by jskeet**
This makes it easier to compare versions. Sample command lines:

```text
--file1=path/to/old.dll --file2=path/to/new.dll
--package=Google.Cloud.Speech.V1 --version1=latest --version2=local
--package=Google.Cloud.Speech.V1 --version1=1.1.0 --version2=1.2.0 --framework=netstandard1.5
```

### CODE REVIEW

File: tools/Google.Cloud.Tools.CompareVersions/CommandLineOptions.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            {
+                Console.Error.WriteLine("--file1 and --version1 cannot be used together");
+                return false;
+            }
+            if (File2 is object && Version2 is object)
+            {
+                Console.Error.WriteLine("--file2 and --version2 cannot be used together");
+                return false;
+            }
+            // We *could* show an error if package or framework as been specified, but file1 and file2 are both specified,
```
Comment (chrisdunelm) [inline tools/Google.Cloud.Tools.CompareVersions/CommandLineOptions.cs]: nit: "as" -> "has"

File: tools/Google.Cloud.Tools.VersionCompat/Google.Cloud.Tools.VersionCompat.csproj [hunk 2]
```diff
@@ -11,6 +10,7 @@
     <PackageReference Include="Mono.Cecil" Version="0.10.3" />
     <PackageReference Include="NuGet.Client" Version="4.2.0" />
```
Comment (chrisdunelm) [inline tools/Google.Cloud.Tools.VersionCompat/Google.Cloud.Tools.VersionCompat.csproj]: Not related to this PR, but is `NuGet.Client` required? The code that loads from nuget doesn't appear to use this?

File: tools/Google.Cloud.Tools.CompareVersions/CommandLineOptions.cs (hunk 1 above)
Comment (jskeet) [reply tools/Google.Cloud.Tools.CompareVersions/CommandLineOptions.cs]: Will fix.

File: tools/Google.Cloud.Tools.VersionCompat/Google.Cloud.Tools.VersionCompat.csproj (hunk 2 above)
Comment (jskeet) [reply tools/Google.Cloud.Tools.VersionCompat/Google.Cloud.Tools.VersionCompat.csproj]: Ooh, good question. Will try it without, and remove in this PR if it's not necessary.

**Pull request merged**